BaseClasses.py

```diff
@@ -1321,30 +1325,30 @@ class Location:
     game: str = "Generic"
     player: int
     name: str
-    address: Optional[int]
-    parent_region: Optional[Region]
+    address: int | None
+    parent_region: Region | None
     locked: bool = False
     show_in_spoiler: bool = True
     progress_type: LocationProgressType = LocationProgressType.DEFAULT
     always_allow: Callable[[CollectionState, Item], bool] = staticmethod(lambda state, item: False)
     access_rule: Callable[[CollectionState], bool] = staticmethod(lambda state: True)
     item_rule: Callable[[Item], bool] = staticmethod(lambda item: True)
-    item: Optional[Item] = None
+    item: Item | None = None
 
-    def __init__(self, player: int, name: str = '', address: Optional[int] = None, parent: Optional[Region] = None):
+    def __init__(self, player: int, name: str = '', address: int | None = None, parent: Region | None = None):
         self.player = player
         self.name = name
         self.address = address
         self.parent_region = parent
 
     def can_fill(self, state: CollectionState, item: Item, check_access: bool = True) -> bool:
         return ((
-            self.always_allow(state, item)
-            and item.name not in state.multiworld.worlds[item.player].options.non_local_items
+                self.always_allow(state, item)
+                and item.name not in state.multiworld.worlds[item.player].options.non_local_items
         ) or (
-            (self.progress_type != LocationProgressType.EXCLUDED or not (item.advancement or item.useful))
-            and self.item_rule(item)
-            and (not check_access or self.can_reach(state))
+                (self.progress_type != LocationProgressType.EXCLUDED or not (item.advancement or item.useful))
+                and self.item_rule(item)
+                and (not check_access or self.can_reach(state))
```

Comment:
I'm not sure if it was intended for these to still be indented by 4 after the change to it.